### PR TITLE
small tutorial rework

### PR DESCRIPTION
**Pull request by cdynak**
there is a little mess in tutorial (HellowW?) project ontology, so i propose two modifications:
- change products names from hello-world to helloworld, registration, chatroom, buddy_status
- change sample project name from multiple forms of 'hello-world' to 'sample'

it would be nice if iphone and android tutorials will have more in common.
maybe it is worth to open new issue here?

### CODE REVIEW

**Comment by bagage** [comment]
Hi again @cdynak, sorry for the delay.
I'd like to merge your PR, but first I need you to sign the [contributor agreement](http://www.belledonne-communications.com/downloads/Belledonne_communications_CA.pdf) and send it back to me please.
Thanks!

**Comment by cdynak** [comment]
I've just send signed contributor agreement to gautier.pelloux-prayer@belledonne-communications.com
hope it's still possible to merge.

**Comment by cdynak** [comment]
hello, how is it going?

**Comment by bagage** [comment]
Hey @cdynak,
Indeed I refactored some stuff after the latest release but forgot your PR, my bad. Would you mind:

1. merge with latest version of `master` branch?
2. merge all your work in a single commit (use `git rebase -i`).

Thanks!

**Comment by cdynak** [comment]
OK, it's merged with today's commit and I've squashed all my changes into one.

**Comment by bagage** [comment]
Hmmm… sorry cdynak but I  actually rethought about what I did and why so to sum up:

1. I deleted the Hello world project because I dislike the idea of having multiple xcode projects to maintain in parallel: indeed anytime the liblinphone SDK change (new library, removal of old library, new framework, etc.), we have to manually link it for each xcode project which is quite burdensome and quite worthless.
2. My plan is to, instead, generate make `cmake` generate a Xcode project automatically for hello world / sample project (hello.c, etc.) so that it will always be up-to-date. But yet this has not been implemented, so I'll do it right now!
Do you think it's fine? There might be alternatives here, so I am open to any suggestion!

tl;dr: I'd like to avoid having to manage multiple Xcode projects using the liblinphone SDK manually.

**Comment by cdynak** [comment]
That's very good idea to keep everything in sync. Less code - less problems.
But IMHO it's also important to make this sample/tutorial project easy accessible and even write something more about it in README.
